configure.ac

```diff
@@ -388,10 +380,58 @@ AS_IF([test "x$NETSTAT_PATH" != "x" ], [
     AC_MSG_NOTICE([NETSTAT_PATH...] $NETSTAT_PATH)
 ])
 
+# use resolvconf if present
+AC_PATH_PROG(RESOLVCONF_PATH, [resolvconf], [DISABLED], "$PATH:/sbin:/usr/sbin")
+
+# allow override at configure time
+AC_ARG_WITH([resolvconf],
+        AS_HELP_STRING([--with-resolvconf],
+                       [Set the path to the resolvconf executable]),
+        RESOLVCONF_PATH="$withval"
+)
+
 AC_SUBST(RESOLVCONF_PATH)
 AS_IF([test "x$RESOLVCONF_PATH" != "x" ], [
     AC_MSG_NOTICE([RESOLVCONF_PATH...] $RESOLVCONF_PATH)
 ])
+AS_IF([test "x$RESOLVCONF_PATH" = "xDISABLED"], [
+	AC_DEFINE(HAVE_RESOLVCONF, 0)
+	AC_MSG_NOTICE([HAVE_RESOLVCONF... 0])
+],[
+	AC_DEFINE(HAVE_RESOLVCONF, 1)
+	AC_MSG_NOTICE([HAVE_RESOLVCONF... 1])
+])
+
+# the default for the --use-resolvconf runtime command line option
+AC_ARG_ENABLE([resolvconf],
+	AS_HELP_STRING([--enable-resolvconf],
+			[Enable usage of resolvconf by default, \
+			use --disable-resolvconf for the opposite]))
+
+# Checks for existence of specific files if not cross-compiling.
+AS_IF([test "x$enable_resolvconf" = "x" ], [
+  AC_CHECK_FILE([$RESOLVCONF_PATH],[
+    AS_IF([$RESOLVCONF_PATH -l &>/dev/null], [
+     enable_resolvconf="yes"
+    ],[
+      AS_IF([ $RESOLVCONF_PATH &>/dev/null ; test $? -eq 99 ], [
```

Comment:
Perhaps document the target distributions where you would expect `resolvconf` to return an exit status of 99?

Reply:
I've found another case (resolvectl acting as resolvconf drop in replacement):
https://www.freedesktop.org/software/systemd/man/resolvconf.html
and we have the resolvconf script which exits with 99 if an unknown command or not enough arguments are supplied https://salsa.debian.org/debian/resolvconf/-/blob/master/bin/resolvconf#L49

Reply:
And that `resolvectl` acting as a drop-in replacement for `resolvconf` is different from the one on _Fedora 31_ which doesn't work?

Reply:
Because on _Fedora_ we have:
```
$ ls -l /usr/sbin/resolvconf
lrwxrwxrwx. 1 root root 17 Feb 11 14:25 /usr/sbin/resolvconf -> ../bin/resolvectl
$ 
$ ls -l /usr/bin/resolvectl 
-rwxr-xr-x. 1 root root 132368 Feb 11 14:25 /usr/bin/resolvectl
$ 
$ rpm -qf /usr/bin/resolvectl
systemd-243.7-1.fc31.x86_64
$ 
$ /usr/bin/resolvectl --help
resolvectl [OPTIONS...] {COMMAND} ...

Send control commands to the network name resolution manager, or
resolve domain names, IPv4 and IPv6 addresses, DNS records, and services.

  -h --help                    Show this help
     --version                 Show package version
     --no-pager                Do not pipe output into a pager
  -4                           Resolve IPv4 addresses
  -6                           Resolve IPv6 addresses
  -i --interface=INTERFACE     Look on interface
  -p --protocol=PROTO|help     Look via protocol
  -t --type=TYPE|help          Query RR with DNS type
  -c --class=CLASS|help        Query RR with DNS class
     --service-address=BOOL    Resolve address for services (default: yes)
     --service-txt=BOOL        Resolve TXT records for services (default: yes)
     --cname=BOOL              Follow CNAME redirects (default: yes)
     --search=BOOL             Use search domains for single-label names
                                                              (default: yes)
     --raw[=payload|packet]    Dump the answer as binary data
     --legend=BOOL             Print headers and additional info (default: yes)

Commands:
  query HOSTNAME|ADDRESS...    Resolve domain names, IPv4 and IPv6 addresses
  service [[NAME] TYPE] DOMAIN Resolve service (SRV)
  openpgp EMAIL@DOMAIN...      Query OpenPGP public key
  tlsa DOMAIN[:PORT]...        Query TLS public key
  status [LINK...]             Show link and server status
  statistics                   Show resolver statistics
  reset-statistics             Reset resolver statistics
  flush-caches                 Flush all local DNS caches
  reset-server-features        Forget learnt DNS server feature levels
  dns [LINK [SERVER...]]       Get/set per-interface DNS server address
  domain [LINK [DOMAIN...]]    Get/set per-interface search domain
  default-route [LINK [BOOL]]  Get/set per-interface default route flag
  llmnr [LINK [MODE]]          Get/set per-interface LLMNR mode
  mdns [LINK [MODE]]           Get/set per-interface MulticastDNS mode
  dnsovertls [LINK [MODE]]     Get/set per-interface DNS-over-TLS mode
  dnssec [LINK [MODE]]         Get/set per-interface DNSSEC mode
  nta [LINK [DOMAIN...]]       Get/set per-interface DNSSEC NTA
  revert LINK                  Revert per-interface configuration

See the resolvectl(1) man page for details.
$ 
```
I thought this `resolvctl` doesn't work, at least not as a ***drop-in*** replacement for``resolvconf`. But it might work with different set of command lline options.

Reply:
OK, this is indeed the man page of the broken `resolvconf`/`resolvctl` on _Fedora_:
https://www.freedesktop.org/software/systemd/man/resolvconf.html

And indeed `resolvctl` is **not** a drop-in replacement for `resolvconf` in our use case. I've created this file:
```
$ cat PIPE_TO_RESOLVCONF 
nameserver xxx.xx.252.252
nameserver xxx.xx.252.202
search xxxx.de xxx.xxxx.de xx.de
```
On _Ubuntu_ 18.04:
```
$ cat /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
# 127.0.0.53 is the systemd-resolved stub resolver.
# run "systemd-resolve --status" to see details about the actual nameservers.

nameserver 127.0.0.53
$ 
$ cat PIPE_TO_RESOLVCONF | /sbin/resolvconf -a "ppp0.openfortivpn"
$ 
$ cat /etc/resolv.conf
# Dynamic resolv.conf(5) file for glibc resolver(3) generated by resolvconf(8)
#     DO NOT EDIT THIS FILE BY HAND -- YOUR CHANGES WILL BE OVERWRITTEN
# 127.0.0.53 is the systemd-resolved stub resolver.
# run "systemd-resolve --status" to see details about the actual nameservers.

nameserver xxx.xx.252.252
nameserver xxx.xx.252.202
nameserver 127.0.0.53
search xxxx.de xxx.xxxx.de xx.de
$ 
```
On _Fedora_ 31:
```
$ cat /etc/resolv.conf 
# Generated by NetworkManager
nameserver 10.0.2.3
$ 
$ cat PIPE_TO_RESOLVCONF | /sbin/resolvconf -a "ppp0.openfortivpn"
$ 
$ cat /etc/resolv.conf 
# Generated by NetworkManager
nameserver 10.0.2.3
$ 
```
I guess the `resolvctl` authors had not thought of the possibility to **pipe** parameters to `resolvconf`. They thought of it as a drop-in replacement with respect to command line options only.

My suggestion:
* In the short term, let's say for 1.12.1, the `resolvconf_fix` branch looks perfect in its current form: on _Fedora_ _openfortivpn_ uses the fallback mechanism.
* In the long term, let's say for 1.13.0, we could support `resolvctl` explicitly in addition to `resolvconf`. In both cases I would recommend using commend line arguments instead of piping into `resolvconf`.